### ISSUE TITLE
Introducing the IPython notebook

### ISSUE THREAD

**Issue by tracykteal** [datacarpentry/python-ecology-lesson#10]
For the R lessons we have an 'Introducing RStudio' module. It would be nice to have an 'Introducing the IPython Notebook' in the Python lesson. 

IPython Notebook itself is a great thing to teach for data analysis, so it seems very worthwhile to teach and use the IPython notebook in these lessons.

**Comment by tracykteal** [comment]
Have these lessons been taught with IPython notebook or with command line Python?

**Comment by lwasser** [comment]
Hey tracy -- we taught these lessons using the notebook. it's definitely convenient BUT its nice for folks to know basic shell to open it. Although with anaconda there is a gui icon that you can use to open it too but then you need to set the working directory.

**Comment by tracykteal** [comment]
Thanks @lwasser. @cmacdonell is going to be teaching @FCTweedie Python NLTK lesson and was thinking about the best way to start, and whether or not to use the notebook. It's good to hear it worked well for you. He is teaching the command line first, I think, so that should help with being able to start anaconda. 

Maybe we should include that setting of the working directory in an 'Introduction to IPython notebook'?

**Comment by tracykteal** [comment]
These tutorials seem good
http://opentechschool.github.io/python-data-intro/core/notebook.html
and from @jdfreder
https://blog.safaribooksonline.com/2013/12/12/start-ipython-notebook/

**Comment by lwasser** [comment]
hey tracy -- 
there is still a very old PR for you to merge but i added a section on setting the working directory here

https://github.com/lwasser/python-ecology/blob/gh-pages/01-starting-with-data.md

If they open the notebook in shell from the director where there data are, then they're good. otherwise they can check it and set it. i think it' sgood in general to check it because sometimes folks forget to open it from a meaningful directory in shell :) 

we definitely did an informal "welcome to ipython notebook!". i think having something like that documented could be great. 
l.

**Comment by tracykteal** [comment]
Thanks I saw that update in your PR. PR looks good. I just had a couple comments. Sorry for the slow response on that. :(

**Comment by lwasser** [comment]
Thanks @tracykteal ! please don't apologize! i completely understand busy schedules!!! :)

**Comment by iamc** [comment]
I did write a ipython-quickstart notebook some time ago for an introduction to scientific Python and plan to use it for my next (and first :-) workshop as an instructor. I may adapt it for these lessons if you want.

https://github.com/iamc/PySS15-scientific-python-tutorial/blob/master/ipython-quickstart.ipynb

**Comment by qjcg** [comment]
Hey @iamc, this looks great! It might be good to add some short challenges (ex: asking learners to create a notebook, run some cells, verify that the notebook has been saved on their system) Also, replacing references to the "IPython notebook" with "Jupyter notebook" would be a forward-looking approach, as the version shipping with anaconda prints the following warning message when `ipython notebook` is run:

```shell
$ ipython notebook
[TerminalIPythonApp] WARNING | Subcommand `ipython notebook` is deprecated and will be removed in future versions.
[TerminalIPythonApp] WARNING | You likely want to use `jupyter notebook` in the future
```

**Comment by iamc** [comment]
Hi @qjcg. So if you are willing to accept something on this line I'll adapt it with your suggestions and also to the lessons template, this as our workshops gets closer and time permits :-)

**Comment by qjcg** [comment]
Awesome, thanks @iamc, please do submit a PR with your adapted lesson when it's ready.

**Comment by iamc** [comment]
OK, I'll do it.

**Comment by djinnome** [comment]
Hi folks, 

  The markdown files in this tutorial can be automagically converted to Jupyter notebooks using  `notedown` . Would folks be interested in a PR that consists of a script that downloads `notedown` and converts these lessons to Jupyter notebook?

**Comment by qjcg** [comment]
Sure @djinnome, if you have a proof of concept script to submit, please do send along a PR!

**Comment by djinnome** [comment]
Hi @qjcg,

   Sure, the pull request is [Port from R to python of Lesson 05 Data visualization with ggplot2](https://github.com/datacarpentry/python-ecology-lesson/pull/96).

 In this PR, I converted the R ecology lesson from markdown to a Jupyter notebook that executes correctly using an R kernel, and then ported the Jupyter notebook that executes correctly using a python kernel.   I then converted the python Jupyter notebook to markdown in order to comply with Data Carpentry standards.
